### PR TITLE
Add Crystal::Repl#parse_and_interpret

### PR DESCRIPTION
**Pull request by bcardiff**
This PR adds a `Crystal::Repl#parse_and_interpret` method that is convenient to use the repl interpreter as library.

The current interpreter api seems lower level . The repl is, as expected, tight to getting the input from stdin.

The added spec shows how the repl can be instantiated and consumed as a library.

### CODE REVIEW

File: src/compiler/crystal/interpreter/value.cr [hunk 1]
```diff
@@ -67,6 +67,15 @@ struct Crystal::Repl::Value
...
   end
 
+  def runtime_type : Crystal::Type
+    if type.is_a?(Crystal::UnionType)
+      type_id = @pointer.as(Int32*).value
+      context.type_from_id(type_id)
+    else
+      type
+    end
+  end
```
Comment (straight-shoota) [inline src/compiler/crystal/interpreter/value.cr:77]: **chore:** This feels like it should be introduced as a separate feature. It's not directly related to `#parse_and_interpret`. The specs should probably work fine without it, but we could also stack PRs.

File: src/compiler/crystal/interpreter/repl.cr [hunk 2]
```diff
@@ -44,6 +42,19 @@ class Crystal::Repl
     end
   end
 
+  record EvalSuccess, value : Value, warnings : WarningCollection
+  record EvalParserError, warnings : WarningCollection
```
Comment (straight-shoota) [inline src/compiler/crystal/interpreter/repl.cr:46]: **question:** Have you considered merging these two types as `EvalResult`? Their shape is similar.

Comment (bcardiff) [reply src/compiler/crystal/interpreter/repl.cr:46]: Sure. Using a nilable value would be enough

File: src/compiler/crystal/interpreter/value.cr (hunk 1 above)
Comment (bcardiff) [reply src/compiler/crystal/interpreter/value.cr:77]: Yup. I aimed to that initially but then I remembered the one repl with prelude restriction and somehow push them together.

I'll send another PR for that.

In it we could use this to show type information as part of the repl: eg `1 :: Int32 <: (String | Int32)` when instructed, but I'm not fully sure about the syntax. Or we can leave that usage for later also.

**Comment by bcardiff** [comment]
I'm not sure if the CI failures are noise or actual failures. I am not familiar with latest changes on it and the missing symbol puzzles me.

**Comment by HertzDevil** [comment]
Either the spec should be moved under `spec/compiler/interpreter/`, or `{% skip_file if flag?(:without_interpreter) %}` should be added to the beginning of the file

**Comment by bcardiff** [comment]
CI is happy. Thanks @HertzDevil

**Review by straight-shoota — APPROVED** [review]

**Pull request merged**